src/backend_model.cc

```diff
@@ -330,6 +330,32 @@ TritonModel::UpdateModelConfig(
   auto outputs_config = config.mutable_output();
   *outputs_config = updated_config.output();
 
+  if (!config.scheduling_choice_case()) {
+    if (updated_config.has_dynamic_batching()) {
+      auto dynamic_batching_config = config.mutable_dynamic_batching();
+      *dynamic_batching_config = updated_config.dynamic_batching();
+    } else if (updated_config.has_sequence_batching()) {
+      auto sequence_batching_config = config.mutable_sequence_batching();
+      *sequence_batching_config = updated_config.sequence_batching();
+    } else if (updated_config.has_ensemble_scheduling()) {
+      auto ensemble_scheduling_config = config.mutable_ensemble_scheduling();
+      *ensemble_scheduling_config = updated_config.ensemble_scheduling();
+    } else {
+      // updated_config is empty
+      config.clear_scheduling_choice();
+    }
+  } else if (
+      config.scheduling_choice_case() !=
+      updated_config.scheduling_choice_case()) {
+    return Status(
+        triton::common::Error::Code::INTERNAL,
+        (std::string("Failed to overwrite scheduling choice, ") +
```

Comment:
Isn't the relation inverse? I would find this to be more clear.
```
Can not update scheduling choice from `config.scheduling_choice_case()` to `updated_config.scheduling_choice_case()` when auto-completing. 
```

Reply:
Fixed to have more readable error message.